src/monster_group.cpp

```diff
@@ -0,0 +1,57 @@
+#include "monster_group.h"
+
+#include <algorithm>
+
+static bool cmp_monster_by_update( const monster &lhs, const monster &rhs ) {
+    return lhs.next_overmap_move() < rhs.next_overmap_move();
+}
+
+void monster_group::add_monster( monster &new_monster ) {
+    // Reset the strongest stimulus since this new monster might not have received it.
```

Comment:
The rationale is expanded a bit on line 28, but I need to bulk up the explanation some. This isn't denying input to the monster, but rather a work-avoidance mechanism.  As stimuli come in, they will be applied to each monster on the list, who will each make their own decision about the stimuli, but if a stronger stimuli is followed by a weaker stimuli, we can avoid applying it.  This is upset by new monsters being inserted into the group though, because we don't have any information about the strongest recent stimuli those monsters experienced, so to be safe I'm just resetting it.
I completely agree that when monsters leave the map, or when they shift from one overmap square to another they should remember their previous goals.

I just realized though, we need to track stimuli of each type separately, if a clearly visible stimuli is followed by a relatively quiet sound, the sound still needs to be applied in case there are blind monsters in the group.